packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -870,14 +870,14 @@ abstract class SliderComponentShape {
     Offset center, {
     required Animation<double> activationAnimation,
     required Animation<double> enableAnimation,
-    bool isDiscrete,
+    bool? isDiscrete,
     required TextPainter labelPainter,
     required RenderBox parentBox,
     required SliderThemeData sliderTheme,
     required TextDirection textDirection,
     required double value,
-    double textScaleFactor,
-    Size sizeWithOverflow,
+    double? textScaleFactor,
+    Size? sizeWithOverflow,
```

Comment:
Same with these. There's also something strange here. Some sub-classes definitely expect these to be non-null...

Reply:
Yeah, I wasn't sure what to do with these. If they are required and non-null then defaults need to be specified for them. It wasn't obvious to me what those defaults should be. I'll talk with Anthony and see if he can give me some guidance here.

Reply:
I cant see any reason for these 3 params not to be `required`

Reply:
K. Thanks for the reply. I will go ahead and update those to `required`.